New Steering Files for Key4hep Workflow

samf25 opened the pull request:
This version of mucoll-benchmarks has entirely native Gaudi algorithms. This will allow for the testing of this workflow with the various geometries.

samf25 committed:
New Workflows

spagangriso commented:
Thanks @samf25 !
One thing from a super quick first look. We want to have here just the steering files, not all the compiled libraries and code files needed. Those will need to go into the software release.
If in need currently to checkout some specific packages, we should have instructions on how to do so in the README.

samf25 commented:
I did it like that because this way is super fast for early testing. There are about 6 or 7 libraries you'd have to check out otherwise and you'd have to set the branches properly. Which would be a lot of work if you're just trying to do an early test. 
I can make it that way if you want, though.

samf25 committed:
Switching to submodule

spagangriso commented:
Thanks Sam.
I think this repo should really only contain the steering files needed with an appropriate image that contains all libraries and their modifications.
To test it easily, we can have a different repo that include this package and others as submodule as usual (e.g. the workspace-like setup) and we can use that for testing, but needs to be separated.

samf25 committed:
SubRepos

samf25 committed:
Remove Libraries

samf25 committed:
move component setup here

samf25 committed:
fix link in readme

samf25 committed:
Update digi_steer.py

samf25 committed:
Update digiAlgList.py

samf25 committed:
components naming abiguity

samf25 committed:
Container paths changed

samf25 committed:
fixed edm4hep pgun

samf25 committed:
Minor changes to get it all up and running

samf25 committed:
Clean up

samf25 committed:
refitter

samf25 committed:
No more submodules

samf25 committed:
No more submodules

samf25 committed:
Update calorimetry_EM.py

samf25 committed:
Update pandora.py

samf25 committed:
Add new_fastJet to algorithm list in recoAlgList.py

samf25 committed:
Add Ability to Set Random Seed

samf25 committed:
Update UniqueIDGenSvc seed from static to dynamic

samf25 committed:
Update import path for new_overlay_full function

samf25 committed:
Name change in new image

samf25 committed:
Revise README for environment setup and library paths

samf25 committed:
Remove Track and RelTrack collections from config

samf25 committed:
Comment out new_fastJet in recoAlgList.py

Comment out the addition of new_fastJet to algList.

samf25 committed:
Remove unused import of new_fastJet

Removed import of new_fastJet from recoAlgList.py.

samf25 committed:
Remove commented FastJetAlg import from pandora.py

samf25 committed:
Add TGeoDescFile to new_CKFTracker initialization

samf25 committed:
Add TGeoDescFile argument to reco_args.py

samf25 committed:
Add TGeoDescFile parameter to new_CKFTracker function

samf25 committed:
Update default file paths to use environment variables

samf25 committed:
Enhance README formatting and instructions

Updated README.md to improve formatting and clarity of environment setup, generation, simulation, digitization, and reconstruction instructions.

samf25 committed:
Revise README for generation and simulation commands

Updated command examples and input/output file specifications in README.

samf25 committed:
Fix parameter value in README generation example

samf25 committed:
Update input and output file formats for simulation

samf25 committed:
Fix input file argument in README

Updated README to correct input file argument syntax for sim.

samf25 committed:
Fix argument passing for new_CKFTracker

samf25 committed:
Add DetectorSchema parameter to CKF_tracking.py

Commit:
Setup script pathing improvement

Commit:
clean up

samf25 committed:
Change MYBUILD path and update PYTHONPATH setup

Updated the MYBUILD variable to point to 'mucoll-benchmarks' instead of 'build'. Added checks for the existence of the 'reconstruction' directory and updated PYTHONPATH accordingly.

samf25 committed:
Geometry controls

samf25 committed:
Better align

samf25 committed:
Better README

samf25 committed:
Clean up

Commit:
Fix pgun script (theta radians) and add phi control

Commit:
pycaches

Commit:
pycaches

spagangriso commented:
Hi @samf25 . I haven't looked in detail at the last commit, just a couple of quick comments:
* for the radians vs degrees, we should ensure this is conistent with the `--help` option. Right now I think it says it should be degrees.
* we should not include python cache files into the git repo (just to be sure)

Thanks!

samf25 commented:
Yep, the help says degrees, but the file was using radians so I just converted in the file so inputting degrees works. And yeah, I accidently pushed the pycaches so I had to remove them

samf25 committed:
MAIA Tracking

samf25 committed:
Change Function Naming Scheme

samf25 committed:
Change default background files number to 812

samf25 committed:
Document Beam-Induced Background (BIB) setup

Added section on Beam-Induced Background (BIB) to README.
